Sming/System/m_printf.cpp

```diff
@@ -245,50 +245,67 @@ void m_printHex(const char* tag, const void* data, size_t len, int addr, size_t
 		return;
 	}
 
+	// tag + ": " + address(*8) + ' ' + hex(*2) + ' ' + char + "\r\n"
+	auto bufSize = taglen + ((bytesPerLine ?: len) * 4) + 16;
+	auto lineBuffer = new char[bufSize];
```

Comment:
Why don't you declare `auto lineBuffer = new char[bufSize];` as `char lineBuffer[bufSize]={0}`? This way the delete on line 310 is not needed and also we avoid potential future  memory leaks when someone decide to put a return statement somewhere in this function without releasing lineBuffer.

Reply:
I my mind variable stack allocations are a bit dodgy. Did a quick search and found this https://www.geeksforgeeks.org/variable-length-arrays-in-c-and-c/

Reply:
Re. memory leaks I can change it to use `std::unique_ptr<>`

Reply:
If I am not mistaken I have seen this recommended in some of the C/C++ code analyzers: 
`char lineBuffer[bufSize]{0}` instead of `auto lineBuffer = new char[bufSize];` when the size of the buffer is constant and will not change.  I cannot remember exactly where but I have seen this but I have found this link: https://rules.sonarsource.com/c/RSPEC-984.

Reply:
The reason I prefer heap allocation here is that passing `bytesPerLine = 0` essentially makes the size unbounded. In general, large/variable memory allocations should be on the heap. Also, don't need to zero-initialise the buffer here.

Reply:
Ok, let it be this way. If this needs to be changed we can alway do it later.

Reply:
Couple of thoughts for a possible future PR:

* Use `String` for the line buffer.
* Add callback parameter for printing each line.
* Re-implement as class elsewhere in framework; number of parameters is unwieldy.
* `m_printHex` can use this class internally (no public dependency on String).